packages/react-components/src/InputNumber.tsx

```diff
@@ -168,12 +168,18 @@ function getValuesFromString (value: string, si: SiDef | null, props: Props): [s
   // Sometimes the value is already formatted, avoid formatting in those cases
   const regex = RegExp('^\\d{1,3}(,\\d{3})*(\\.\\d+)?$');
   // Format only if required
-  if (!regex.exec(value)) {
-    const defaultValue = valueBn
-      ? formatBalance(valueBn, { forceUnit: '-', withSi: false }).replace(',', ',')
-      : valueBn;
-    formatedValue = defaultValue.split('.')[0];
+  if (value && !regex.exec(value)) {
+    let decimalValue;
+    if (value.includes('.')) {
+      decimalValue = value.split('.');
+      value = decimalValue[0];
+    }
+    formatedValue = formatNumber(new BN(value.replace(/,/g, '')));
+    formatedValue = decimalValue ? `${formatedValue}.${decimalValue[1]}` : formatedValue;
   }
```

Comment:
It would be better not to reassign value to local variable if possible, I guess that's one reason why `const` is suggested over `let` , how about something like below:

```js
const valueAsArray = value.split(".");
formatedValue = valueAsArray[1]
  ? `${formatNumber(new BN(valueAsArray[0].replace(/,/g, "")))}.${
      valueAsArray[1]
    }`
  : valueAsArray[0];
```

Reply:
Here, if user enters 113. then valueAsArray[1] would not exist. but if users enters something before '.' decimal (won't happen generally) lets say it becomes 4113. here we still want to format it as 4,113.

Reply:
Right, the line of `: valueAsArray[0]` should have been like `: formatNumber(new BN(valueAsArray[0].replace(/,/g, "")))`, but not a big deal, depends on you, just some code refactoring suggestions.

Reply:
In that case for some value like 1334., it will become 1334 and decimal will disappear.